fastkafka/_application/app.py

```diff
@@ -784,3 +796,58 @@ def sync_inner(
             for name, (f, producer, kwargs) in self._producers_store.items()
         }
     )
+
+# %% ../../nbs/015_FastKafka.ipynb 56
+@patch
+def benchmark(
+    self: FastKafka,
+    interval: Union[int, timedelta] = 1,
+    *,
+    sliding_window_size: Optional[int] = None,
+) -> Callable[[Any], Any]:
+    """Decorator to benchmark produces/consumes functions
+
+    Args:
+        interval: Period to use to calculate throughput. If value is of type int,
+            then it will be used as seconds. If value is of type timedelta,
+            then it will be used as it is. default: 1 - one second
+        sliding_window_size: The size of the sliding window to use to calculate
+            average throughput. default: None - By default average throughput is
+            not calculated
+    """
+
+    def _decorator(func: Callable[[Any], Any]) -> Callable[[Any], Any]:
```

Comment:
typing is close, but not close enough. Use TypeVar to ensure that you return the same function type as you are receiving as input.